### PR TITLE
Add `mc encrypt` command

### PR DESCRIPTION
**Pull request by poornas**

### CODE REVIEW

File: cmd/bucket-encrypt.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ 
+ USAGE:
+   {{.HelpName}} TARGET [info | disable | [SSE-S3 | SSE-KMS kms-key]]
+ 
+ FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+ EXAMPLES:
+   1. Display bucket encryption status for bucket "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket info
```
Comment (kannappanr) [inline cmd/bucket-encrypt.go:48]:
```suggestion
 {{.Prompt}} {{.HelpName}} myminio/mybucket info
```

File: cmd/bucket-encrypt.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+   {{.HelpName}} - {{.Usage}}
+ 
+ USAGE:
+   {{.HelpName}} TARGET [info | disable | [SSE-S3 | SSE-KMS kms-key]]
+ 
+ FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+ EXAMPLES:
+   1. Display bucket encryption status for bucket "mybucket".
```
Comment (kannappanr) [inline cmd/bucket-encrypt.go:47]: Please fix the format here

File: cmd/bucket-encrypt.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+// checkBucketEncryptSyntax - validate all the passed arguments
+func checkBucketEncryptSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) < 2 || len(ctx.Args()) > 3 {
+		cli.ShowCommandHelpAndExit(ctx, "encrypt", 1) // last argument is exit code
+	}
+}
+
+type bucketEncryptMessage struct {
+	Op         string
```
Comment (vadmeste) [inline cmd/bucket-encrypt.go:66]: `json:"op"`

File: cmd/client-s3.go [hunk 4]
```diff
@@ -2313,3 +2315,55 @@ func (c *S3Client) SetVersioning(ctx context.Context, status string) *probe.Erro
...
+
+	config, e := c.api.GetBucketEncryption(ctx, bucket)
+	if e != nil {
+		return "", "", probe.NewError(e)
+	}
+	for _, rule := range config.Rules {
+		algorithm = rule.Apply.SSEAlgorithm
+		if rule.Apply.KmsMasterKeyID != "" {
+			keyID = rule.Apply.KmsMasterKeyID
+			break
```
Comment (vadmeste) [inline cmd/client-s3.go:2334]: I do not understand this exactly, are we supposed to have only one rule for now ?

File: cmd/client-s3.go [hunk 5]
```diff
@@ -2313,3 +2315,55 @@ func (c *S3Client) SetVersioning(ctx context.Context, status string) *probe.Erro
...
+}
+
+// SetEncryption - Set encryption configuration on a bucket
+func (c *S3Client) SetEncryption(ctx context.Context, encType string, kmsKeyID string) *probe.Error {
+	bucket, _ := c.url2BucketAndObject()
+	if bucket == "" {
+		return probe.NewError(BucketNameEmpty{})
+	}
+	var config *sse.Configuration
+	switch strings.ToLower(encType) {
```
Comment (vadmeste) [inline cmd/client-s3.go:2347]: Let's error out in case of default (I know it is just a defensive code)

File: cmd/bucket-encrypt.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	Status     string `json:"status"`
+	URL        string `json:"url"`
+	Encryption struct {
+		Algorithm string `json:"algorithm"`
+		KeyID     string `json:"keyId"`
```
Comment (kannappanr) [inline cmd/bucket-encrypt.go:71]:
```suggestion
		KeyID     string `json:"keyId,omitempty"`
```

File: cmd/bucket-encrypt.go (hunk 3 above)
Comment (kannappanr) [inline cmd/bucket-encrypt.go:66]:
```suggestion
	Op         string `json:"op"`
```

File: cmd/bucket-encrypt.go [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+	if len(ctx.Args()) < 2 || len(ctx.Args()) > 3 {
+		cli.ShowCommandHelpAndExit(ctx, "encrypt", 1) // last argument is exit code
+	}
+}
+
+type bucketEncryptMessage struct {
+	Op         string
+	Status     string `json:"status"`
+	URL        string `json:"url"`
+	Encryption struct {
```
Comment (kannappanr) [inline cmd/bucket-encrypt.go:69]: This also needs to be lower case in `json`

File: cmd/client-s3.go (hunk 4 above)
Comment (poornas) [reply cmd/client-s3.go:2334]: yes

File: cmd/client-s3.go (hunk 5 above)
Comment (poornas) [reply cmd/client-s3.go:2347]: done

File: cmd/bucket-encrypt.go [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+type bucketEncryptMessage struct {
+	Op         string `json:"op"`
+	Status     string `json:"status"`
+	URL        string `json:"url"`
+	Encryption struct {
+		Algorithm string `json:"algorithm"`
+		KeyID     string `json:"keyId,omitempty"`
+	} `json:"encryption"`
```
Comment (kannappanr) [inline cmd/bucket-encrypt.go:76]:
```suggestion
	} `json:"encryption, omitempty"`
```

Comment (poornas) [reply cmd/bucket-encrypt.go:76]: don't think this should be omitted - if there is no encryption configuration the struct should be empty

Comment (kannappanr) [reply cmd/bucket-encrypt.go:76]: no, this is required for the clear command
```
$ mc bucket encrypt play/kannappan clear --json
{
 "op": "clear",
 "status": "success",
 "url": "play/kannappan",
 "encryption": {
  "algorithm": ""
 }
}
```

File: cmd/bucket-encrypt.go [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+   1. Display bucket encryption status for bucket "mybucket".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket info
+
+   2. Enable SSE-S3 auto encryption on bucket "mybucket" for alias "myminio".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket sse-s3
+   
+   3. Enable SSE-KMS auto encryption with kms key on bucket "mybucket" for alias "s3".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket sse-kms arn:aws:kms:us-east-1:xxx:key/xxx
+ 
+
```
Comment (vadmeste) [inline cmd/bucket-encrypt.go:56]: Remove one line here

File: cmd/bucket-encrypt.go [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+      {{.Prompt}} {{.HelpName}} myminio/mybucket info
+
+   2. Enable SSE-S3 auto encryption on bucket "mybucket" for alias "myminio".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket sse-s3
+   
+   3. Enable SSE-KMS auto encryption with kms key on bucket "mybucket" for alias "s3".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket sse-kms arn:aws:kms:us-east-1:xxx:key/xxx
+ 
+ 
+   3. Remove auto encryption config on bucket "mybucket" for alias "myminio".
```
Comment (vadmeste) [inline cmd/bucket-encrypt.go:57]: s/3/4

**Comment by poornas** [comment]
Please re-review - the command changed to move encrypt to top level

**Pull request merged**